packages/website/blog/2022-09-03-automated-rule-docs-with-docusaurus-and-remark.md

```diff
@@ -0,0 +1,224 @@
+Each of those rule docs pages includes a description of the rule, any options it allows, links to its source code, and other important information about its usage.
+
+Until recently, keeping descriptions of rules consistent between their source code and docs pages was a cumbersome manual chore.
+We'd written a suite of Jest tests to verify they matched -- but those tests didn't capture everything, often failed with cryptic messages, and were missed by less experienced contributors.
+
+We're happy to say that now, we've overhauled rule docs pages to automatically generate metadata information from rule source code.
+That means the pages always display up-to-date information without developers needing to manually rewrite docs on rule changes.
+Hooray! 🎉
+
+This blog post gives an overview of the [chore(website): generate rule docs options automatically](https://github.com/typescript-eslint/typescript-eslint/pull/5386).
+
+<!--truncate-->
+
+## Tools In Play
+
+https://typescript-eslint.io is built on [Docusaurus](https://docusaurus.io), a powerful static site generator tailored to documentation websites and blogs.
+Docusaurus comes with sensible defaults, extensive configuration options, and thoroughly fleshed out documentation.
+We're big fans.
+❤️
+
+[Docusaurus blogs](https://docusaurus.io/docs/blog) support writing blog posts in [MDX](https://mdxjs.com), a rich superset of Markdown that allows inserting JSX components.
+Docusaurus supports MDX using [remark](https://remark.js.org), a Markdown processor powered by plugins.
+Remark plugins take in Markdown files in a format known as an Abstract Syntax Trees, or AST, and output modified versions of those ASTs.
+
+You can see what Remark's AST equivalent of a Markdown document looks like by visiting [AST Explorer](https://astexplorer.net) and selecting the Markdown language.
+
+![astexplorer.net showing the default snippet with the paragraph highlighted](./ast-explorer-remark.png)
+
+> _astexplorer.net showing the default snippet with the paragraph highlighted_
+
+Lastly, Remark and several other packages around parsing and modifying ASTs are built on the [`unified` project](https://github.com/unifiedjs/unified).
+The unified AST format adheres to the [`unist` specification](https://github.com/syntax-tree/unist).
+
+### Custom Remark Plugins
+
+Remark allows projects to register any number of custom Remark plugins for transforming Markdown contents.
+For example, typescript-eslint.io already uses [`remark-docusaurus-tabs`](https://www.npmjs.com/package/remark-docusaurus-tabs)
+
+The relevant parts of [typescript-eslint.io's Docusaurus config](https://github.com/typescript-eslint/typescript-eslint/blob/39829c01906f326fec94e9b3a5fdb1730eb02002/packages/website/docusaurusConfig.ts) specify the `remark-docusaurus-tabs` plugin and custom `./plugins/generated-rule-docs` plugins, both with their default options (`{}`):
```

Comment:
```suggestion
The relevant parts of [typescript-eslint.io's Docusaurus config](https://github.com/typescript-eslint/typescript-eslint/blob/39829c01906f326fec94e9b3a5fdb1730eb02002/packages/website/docusaurusConfig.ts) specify the `remark-docusaurus-tabs` plugin and the custom `./plugins/generated-rule-docs` plugin, both with their default options (`{}`):
```

(Also, the default option is technically optional, but it's fine if you want to be explicit about it)

Reply:
Aha, nice - I'll just remove the unnecessary `{}`s.